codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonRpcProtocolGenerator.java

```diff
@@ -0,0 +1,66 @@
+package software.amazon.smithy.aws.go.codegen;
+
+import java.util.Set;
+import software.amazon.smithy.codegen.core.CodegenException;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoStackStepMiddlewareGenerator;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.integration.HttpRpcProtocolGenerator;
+import software.amazon.smithy.go.codegen.integration.ProtocolGenerator;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.Shape;
+
+/**
+ * Handles generating the aws.rest-json protocol for services.
+ *
+ * @inheritDoc
+ *
+ * @see HttpRpcProtocolGenerator
+ */
+abstract class JsonRpcProtocolGenerator extends HttpRpcProtocolGenerator {
```

Comment:
This class, or the HttpRpcProtocolGenerator also want to implement the protocol tests

```java
    @Override
    public void generateProtocolTests(GenerationContext context) {
        new HttpProtocolTestGenerator(context,
                new HttpProtocolUnitTestRequestGenerator.Builder(),
                new HttpProtocolUnitTestResponseGenerator.Builder(),
                new HttpProtocolUnitTestResponseErrorGenerator.Builder()
        ).generateProtocolTests();
    }
```